core/src/main/java/org/frankframework/parameters/ParameterValue.java

```diff
@@ -157,6 +160,12 @@ private String valueAsString() {
 				throw new IllegalStateException("cannot open stream", e);
 			}
 		}
+
+		if (getDefinition() instanceof DateParameter dateParameter && value instanceof Date date) {
+			final DateFormat df = new SimpleDateFormat(dateParameter.getFormatString());
+			return df.format(date);
+		}
```

Comment:
I think a little cleanup is required here. The methods `asIntegerValue` an `asBooleanValue` should get something like:
```
if(value instanceof Integer i) {
	return i;
}
```
to avoid back and forth conversions.

The methods byte, double, long, float, short can go, as they are not used and I honestly don't know if this would ever be useful. (perhaps long ..?)

A little test would really come in handy here! eg. What happens if you use a NumberParameter and do asBoolean.